snapshot-labs/sx-monorepo
# fix: always show voting type selection in Editor

### Summary



Closes: #509 #516

Summary:

- feat(ui): always show voting type selection in proposal editor.
- bug: voting type settings set at space being ignored

This PR will resolves #509 by:

- Always showing the voting type selection on the proposal creation page, even if there's only one choice (in which case, the voting type selection form will be disabled)

This PR wil resolves #516 by:

- Add a new `voting_types` (`VoteType[]`) property to `Space`, which will hold the whitelist of voting type for the given space. Offchain will get the list from the graphql results, whereas onchains's one will be set from a constant 
- Edit the Editor to use the dynamic list of votingTypes from the space => Most of the changed files are from this update, as creating a draft is now an async task, since it has to load the vote type from the space settings

### How to test

1. Go to a proposal creation page on onchain space
2. It should show the voting type selector on the right sidebar, locked to `Basic Choice`.
3. Go to http://localhost:8080/#/s:ens.eth/create
4. You can choose from a list of voting type, with `Single Choice` being the default selection
5. Go to http://localhost:8080/#/s:safe.eth/create
6. The voting type selection should be locked to `Single Choice`, as it's the type set by the space
7. Draft should be created only if there's a change in the form
